--- iscsitarget-1.4.20.3+svn502_org/kernel/nthread.c	2016-06-13 23:35:35.132826492 -0700
+++ iscsitarget-1.4.20.3+svn502/kernel/nthread.c	2016-06-13 23:41:27.605203377 -0700
@@ -126,7 +126,7 @@ static int do_recv(struct iscsi_conn *co
 
 	oldfs = get_fs();
 	set_fs(get_ds());
-	res = sock_recvmsg(conn->sock, &msg, len, MSG_DONTWAIT | MSG_NOSIGNAL);
+	res = sock_recvmsg(conn->sock, (struct msghdr *)&msg, len, MSG_DONTWAIT | MSG_NOSIGNAL);
 	set_fs(oldfs);
 
 	if (res <= 0) {
@@ -478,7 +478,7 @@ static int tx_ddigest(struct iscsi_cmnd
 	iov.iov_base = (char *) (&cmnd->ddigest) + (sizeof(u32) - rest);
 	iov.iov_len = rest;
 
-	res = kernel_sendmsg(cmnd->conn->sock, &msg, &iov, 1, rest);
+	res = kernel_sendmsg(cmnd->conn->sock, (struct msghdr *) &msg, &iov, 1, rest);
 
 	if (res > 0) {
 		cmnd->conn->write_size -= res;
@@ -677,7 +677,7 @@ static void close_conn(struct iscsi_conn
 
 	write_lock_bh(&conn->sock->sk->sk_callback_lock);
 	conn->sock->sk->sk_state_change = target->nthread_info.old_state_change;
-	conn->sock->sk->sk_data_ready = target->nthread_info.old_data_ready;
+	conn->sock->sk->sk_data_ready = (void (*) (struct sock *))target->nthread_info.old_data_ready;
 	conn->sock->sk->sk_write_space = target->nthread_info.old_write_space;
 	write_unlock_bh(&conn->sock->sk->sk_callback_lock);
 
--- iscsitarget-1.4.20.3+svn502_org/kernel/block-io.c	2016-06-13 23:35:35.255833998 -0700
+++ iscsitarget-1.4.20.3+svn502/kernel/block-io.c	2016-06-13 23:49:39.379916524 -0700
@@ -57,7 +57,6 @@ static int
 blockio_make_request(struct iet_volume *volume, struct tio *tio, int rw)
 {
 	struct blockio_data *bio_data = volume->private;
-	struct request_queue *bdev_q = bdev_get_queue(bio_data->bdev);
 	struct tio_work *tio_work;
 	struct bio *tio_bio = NULL, *bio = NULL, *biotail = NULL;
 	struct blk_plug plug;
@@ -69,6 +68,7 @@ blockio_make_request(struct iet_volume *
 	int err = 0;
 	loff_t ppos = tio->offset;	
 #else	
+	struct request_queue *bdev_q = bdev_get_queue(bio_data->bdev);
 	int max_pages = 1;
 	int err = 0;
 
@@ -102,7 +102,7 @@ blockio_make_request(struct iet_volume *
 		/* bi_sector is ALWAYS in units of 512 bytes */
 		bio->bi_iter.bi_sector = ppos >> 9;
 		bio->bi_bdev = bio_data->bdev;
-		bio->bi_end_io = blockio_bio_endio;
+		bio->bi_end_io = (bio_end_io_t *) blockio_bio_endio;
 		bio->bi_private = tio_work;
 
 		if (tio_bio)
--- iscsitarget-1.4.20.3+svn502_org/kernel/conn.c	2016-06-13 23:35:35.193830215 -0700
+++ iscsitarget-1.4.20.3+svn502/kernel/conn.c	2016-06-13 23:43:06.932203203 -0700
@@ -140,8 +140,8 @@ static void iet_socket_bind(struct iscsi
 	target->nthread_info.old_state_change = conn->sock->sk->sk_state_change;
 	conn->sock->sk->sk_state_change = iet_state_change;
 
-	target->nthread_info.old_data_ready = conn->sock->sk->sk_data_ready;
-	conn->sock->sk->sk_data_ready = iet_data_ready;
+	target->nthread_info.old_data_ready = (void (*) (struct sock *, int )) conn->sock->sk->sk_data_ready;
+	conn->sock->sk->sk_data_ready = (void (*) (struct sock *)) iet_data_ready;
 
 	target->nthread_info.old_write_space = conn->sock->sk->sk_write_space;
 	conn->sock->sk->sk_write_space = iet_write_space;
