--- a/Source/JavaScriptCore/wtf/dtoa/utils.h	2013-07-18 12:07:49.511804706 +0530
+++ b/Source/JavaScriptCore/wtf/dtoa/utils.h	2013-07-18 12:09:07.655807748 +0530
@@ -46,7 +46,7 @@
 // disabled.)
 // On Linux,x86 89255e-22 != Div_double(89255.0/1e22)
 #if defined(_M_X64) || defined(__x86_64__) || \
-defined(__ARMEL__) || defined(__avr32__) || \
+defined(__ARMEL__) || defined(__ARMEB__) || defined(__avr32__) || \
 defined(__hppa__) || defined(__ia64__) || \
 defined(__mips__) || defined(__powerpc__) || \
 defined(__sparc__) || defined(__s390__) || \
