Comment out the doc (un)installing statement in pmdk

Upstream-Status: Local
Signed-off-by: Harish Kumar Gurram <hgurram@mvista.com>

diff --git a/Makefile b/Makefile
index 7f2417d..83972fd 100644
--- a/Makefile
+++ b/Makefile
@@ -142,7 +142,6 @@ rpm dpkg: pkg-clean source
 
 install uninstall:
 	$(MAKE) -C src $@
-	$(MAKE) -C doc $@
 
 .PHONY: all clean clobber test check cstyle check-license install uninstall\
 	source rpm dpkg pkg-clean pcheck check-remote format doc $(SUBDIRS)
