Upstream-Status: Local
Signed-off-by: Harish Kumar Gurram <hgurram@mvista.com>

diff --git a/src/jemalloc/Makefile.in b/src/jemalloc/Makefile.in
index 04a3806..99e7f89 100644
--- a/src/jemalloc/Makefile.in
+++ b/src/jemalloc/Makefile.in
@@ -309,14 +309,18 @@ install_doc_html:
 	install -d $(DATADIR)/doc/jemalloc$(install_suffix)
 	@for d in $(DOCS_HTML); do \
 	echo "install -m 644 $$d $(DATADIR)/doc/jemalloc$(install_suffix)"; \
-	install -m 644 $$d $(DATADIR)/doc/jemalloc$(install_suffix); \
+	if [ -e $$d ] ; then \
+		install -m 644 $$d $(DATADIR)/doc/jemalloc$(install_suffix); \
+	fi;\
 done
 
 install_doc_man:
 	install -d $(MANDIR)/man3
 	@for d in $(DOCS_MAN3); do \
 	echo "install -m 644 $$d $(MANDIR)/man3"; \
-	install -m 644 $$d $(MANDIR)/man3; \
+	if [ -e $$d ] ; then \
+		install -m 644 $$d $(MANDIR)/man3; \
+	fi;\
 done
 
 install_doc: install_doc_html install_doc_man
