diff --git a/sample/Makefile.am b/sample/Makefile.am
index c926f4e..40edafe 100644
--- a/sample/Makefile.am
+++ b/sample/Makefile.am
@@ -9,7 +9,9 @@ AUTOMAKE_OPTIONS = foreign no-dependencies
 LDADD = $(LIBEVENT_GC_SECTIONS) ../libevent.la
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/compat -I$(top_srcdir)/include -I../include
 
-noinst_PROGRAMS = event-test time-test signal-test dns-example hello-world http-server
+SAMPLEdir = $(datadir)/$(PACKAGE)-$(VERSION)/samples
+
+SAMPLE_PROGRAMS = event-test time-test signal-test dns-example hello-world http-server
 
 event_test_SOURCES = event-test.c
 time_test_SOURCES = time-test.c
@@ -19,7 +21,7 @@ hello_world_SOURCES = hello-world.c
 http_server_SOURCES = http-server.c
 
 if OPENSSL
-noinst_PROGRAMS += le-proxy
+SAMPLE_PROGRAMS += le-proxy
 le_proxy_SOURCES = le-proxy.c
 le_proxy_LDADD = $(LDADD) ../libevent_openssl.la -lssl -lcrypto ${OPENSSL_LIBADD}
 endif
diff --git a/test/Makefile.am b/test/Makefile.am
index b10c41a..7ccc49e 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -10,11 +10,13 @@ AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/compat -I$(top_srcdir)/include -I.
 
 EXTRA_DIST = regress.rpc regress.gen.h regress.gen.c rpcgen_wrapper.sh test.sh
 
-noinst_PROGRAMS = test-init test-eof test-weof test-time \
+TESTdir = $(datadir)/$(PACKAGE)-$(VERSION)/tests
+
+TEST_PROGRAMS = test-init test-eof test-weof test-time \
 	bench bench_cascade bench_http bench_httpclient test-ratelim \
 	test-changelist
 if BUILD_REGRESS
-noinst_PROGRAMS += regress
+TEST_PROGRAMS += regress
 endif
 EXTRA_PROGRAMS = regress
 noinst_HEADERS = tinytest.h tinytest_macros.h regress.h tinytest_local.h
