Upstream-Status: Inappropriate [MV specific]

From 7fb80314259133cdeb3128846d53077bc78ccc27 Mon Sep 17 00:00:00 2001
From: "Jeremy A. Puhlman" <jpuhlman@mvista.com>
Date: Thu, 19 May 2022 17:09:51 +0000
Subject: [PATCH 4/7] Ignore invalid regnum in DM_CFA offset extended

---
 src/dwarf/Gparser.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/src/dwarf/Gparser.c b/src/dwarf/Gparser.c
index da170d4..6801220 100644
--- a/src/dwarf/Gparser.c
+++ b/src/dwarf/Gparser.c
@@ -189,10 +189,18 @@ run_cfi_program (struct dwarf_cursor *c, dwarf_state_record_t *sr,
           break;
 
         case DW_CFA_offset_extended:
-          if (((ret = read_regnum (as, a, addr, &regnum, arg)) < 0)
-              || ((ret = dwarf_read_uleb128 (as, a, addr, &val, arg)) < 0))
+	      if ((ret = dwarf_read_uleb128 (as, a, addr, &val, arg)) < 0)    
             break;
-          set_reg (sr, regnum, DWARF_WHERE_CFAREL, val * dci->data_align);
+          regnum = val;
+          if ((ret = dwarf_read_uleb128 (as, a, addr, &val, arg)) < 0)
+            break;
+          /* If the register number is invalid then ignore it,
+           * but continue anyway. */
+          if (regnum >= DWARF_NUM_PRESERVED_REGS)
+            Debug (1, "Invalid register number %u in DW_CFA_offset_extended\n",
+                   (unsigned int) regnum);
+          else
+            set_reg (sr, regnum, DWARF_WHERE_CFAREL, val * dci->data_align);
           Debug (15, "CFA_offset_extended r%lu at cf+0x%lx\n",
                  (long) regnum, (long) (val * dci->data_align));
           break;
-- 
2.35.3

