diff -Naurp source_org/config/Makefile.inc.in source/config/Makefile.inc.in
--- source_org/config/Makefile.inc.in	2015-07-31 00:43:40.707647569 +0530
+++ source/config/Makefile.inc.in	2015-07-31 00:44:08.383681127 +0530
@@ -188,7 +188,7 @@ AR = @AR@
 CC = @CC@
 CPP = @CPP@
 CFLAGS = @UCONFIG_CFLAGS@ 
-CPPFLAGS = @UCONFIG_CPPFLAGS@ -I$(prefix)/include
+CPPFLAGS = @UCONFIG_CPPFLAGS@
 CXXFLAGS = @UCONFIG_CXXFLAGS@
 CXX = @CXX@
 DEFAULT_MODE = @DATA_PACKAGING_MODE@
diff -Naurp source_org/config/icu-config-bottom source/config/icu-config-bottom
--- source_org/config/icu-config-bottom	2015-07-31 01:20:26.938279130 +0530
+++ source/config/icu-config-bottom	2015-07-31 01:23:14.170421305 +0530
@@ -238,7 +238,7 @@ do
 	    ;;
 
 	--cppflags-searchpath)
-	    echo $ECHO_N "-I${prefix}/include ${ECHO_C}"
+	    echo $ECHO_N "${CPPFLAGS} ${ECHO_C}"
 	    ;;
 
 	--cppflags-dynamic)
@@ -283,7 +283,7 @@ do
 	    ;;
 
 	--ldflags-searchpath)
-	    echo $ECHO_N "-L${libdir} ${ECHO_C}"
+	    echo $ECHO_N "${LDFLAGS} ${ECHO_C}"
 	    ;;
 
 	--detect-prefix)
