Source: http://lists.infradead.org/pipermail/kexec/2016-December/017804.html

diff --git a/kexec/arch/arm/Makefile b/kexec/arch/arm/Makefile
index 871979e0ae26..4454f47a48df 100644
--- a/kexec/arch/arm/Makefile
+++ b/kexec/arch/arm/Makefile
@@ -27,9 +27,8 @@ arm_KEXEC_SRCS += $(libfdt_SRCS)
 
 arm_UIMAGE = kexec/kexec-uImage.c
 arm_PHYS_TO_VIRT = kexec/arch/arm/phys_to_virt.c
-arm_PHYS_TO_VIRT += kexec/arch/arm/iomem.h
-arm_PHYS_TO_VIRT += kexec/arch/arm/phys_to_virt.h
 
 dist += kexec/arch/arm/Makefile $(arm_KEXEC_SRCS) $(arm_PHYS_TO_VIRT)	\
-	kexec/arch/arm/crashdump-arm.h kexec/arch/arm/kexec-arm.h	\
+	kexec/arch/arm/iomem.h kexec/arch/arm/phys_to_virt.h \
+	kexec/arch/arm/crashdump-arm.h kexec/arch/arm/kexec-arm.h \
 	kexec/arch/arm/include/arch/options.h
