diff -Naurp webkitgtk-2.8.5_org/Source/WTF/wtf/dtoa/utils.h webkitgtk-2.8.5/Source/WTF/wtf/dtoa/utils.h
--- webkitgtk-2.8.5_org/Source/WTF/wtf/dtoa/utils.h	2015-12-07 23:49:41.466771892 -0800
+++ webkitgtk-2.8.5/Source/WTF/wtf/dtoa/utils.h	2015-12-07 23:47:05.644786276 -0800
@@ -46,7 +46,11 @@
 // disabled.)
 // On Linux,x86 89255e-22 != Div_double(89255.0/1e22)
 #if defined(_M_X64) || defined(__x86_64__) || \
-defined(__ARMEL__) || \
+defined(__ARMEL__) || defined(__ARMEB__) || defined(__avr32__) || \
+defined(__hppa__) || defined(__ia64__) || \
+defined(__mips__) || defined(__powerpc__) || \
+defined(__sparc__) || defined(__s390__) || \
+defined(__SH4__) || defined(__alpha__) || \
 defined(_MIPS_ARCH_MIPS32R2)
 #define DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS 1
 #elif CPU(MIPS) || CPU(MIPS64) || CPU(PPC) || CPU(PPC64) || CPU(PPC64LE) || CPU(SH4) || CPU(S390) || CPU(S390X) || CPU(IA64) || CPU(ALPHA) || CPU(ARM64) || CPU(HPPA)
