Index: lxc-0.9.0/src/lxc/conf.c
===================================================================
--- lxc-0.9.0.orig/src/lxc/conf.c
+++ lxc-0.9.0/src/lxc/conf.c
@@ -1860,7 +1860,10 @@ static int setup_netdev(struct lxc_netde
 
 	/* rename the interface name */
 	err = lxc_netdev_rename_by_name(ifname, netdev->name);
-	if (err) {
+	if ( (netdev->type == LXC_NET_PHYS || netdev->type == LXC_NET_VLAN)
+			&& err == -EEXIST) {
+		DEBUG("using existing interface '%s'", netdev->name);
+	} else if (err) {
 		ERROR("failed to rename %s->%s : %s", ifname, netdev->name,
 		      strerror(-err));
 		return -1;
@@ -2276,14 +2279,21 @@ static int instanciate_vlan(struct lxc_h
 		return -1;
 	}
 
-	err = snprintf(peer, sizeof(peer), "vlan%d", netdev->priv.vlan_attr.vid);
+	if (!netdev->name)
+		err = snprintf(peer, sizeof(peer), "vlan%d",
+			       netdev->priv.vlan_attr.vid);
+	else
+		err = snprintf(peer, sizeof(peer), "%s.%d",
+			       netdev->link, netdev->priv.vlan_attr.vid);
 	if (err >= sizeof(peer)) {
 		ERROR("peer name too long");
 		return -1;
 	}
 
 	err = lxc_vlan_create(netdev->link, peer, netdev->priv.vlan_attr.vid);
-	if (err) {
+	if (err == -EEXIST) {
+		DEBUG("using existing vlan interface '%s'", peer);
+	} else if (err) {
 		ERROR("failed to create vlan interface '%s' on '%s' : %s",
 		      peer, netdev->link, strerror(-err));
 		return -1;
@@ -2296,8 +2306,8 @@ static int instanciate_vlan(struct lxc_h
 		return -1;
 	}
 
-	DEBUG("instanciated vlan '%s', ifindex is '%d'", " vlan1000",
-	      netdev->ifindex);
+	DEBUG("instanciated vlan '%s', ifindex is '%d'",
+	      peer, netdev->ifindex);
 
 	return 0;
 }
