Index: lxc-0.9.0/doc/lxc.conf.sgml.in
===================================================================
--- lxc-0.9.0.orig/doc/lxc.conf.sgml.in
+++ lxc-0.9.0/doc/lxc.conf.sgml.in
@@ -100,6 +100,10 @@ Foundation, Inc., 59 Temple Place, Suite
 	      <option>i686</option>,
 	      <option>x86_64</option>,
 	      <option>amd64</option>
+	      <option>ppc</option>
+	      <option>ppc64</option>
+	      <option>mips</option>
+	      <option>mips64</option>
 	    </para>
 	  </listitem>
 	</varlistentry>
Index: lxc-0.9.0/src/lxc/confile.c
===================================================================
--- lxc-0.9.0.orig/src/lxc/confile.c
+++ lxc-0.9.0/src/lxc/confile.c
@@ -1470,11 +1470,15 @@ signed long lxc_config_parse_arch(const
 	struct per_name {
 		char *name;
 		unsigned long per;
-	} pername[4] = {
+	} pername[8] = {
 		{ "x86", PER_LINUX32 },
 		{ "i686", PER_LINUX32 },
 		{ "x86_64", PER_LINUX },
 		{ "amd64", PER_LINUX },
+		{ "ppc", PER_LINUX32 },
+		{ "ppc64", PER_LINUX },
+		{ "mips", PER_LINUX32 },
+		{ "mips64", PER_LINUX },
 	};
 	size_t len = sizeof(pername) / sizeof(pername[0]);
 
