diff --git a/LVM2.2.02.97/scripts/lvmdump.sh.old b/LVM2.2.02.97/scripts/lvmdump.sh
index 5e764f2..1cdcfb5 100755
--- LVM2.2.02.97/scripts/lvmdump.sh.old
+++ LVM2.2.02.97/scripts/lvmdump.sh
@@ -80,7 +80,7 @@ while getopts :acd:hmu opt; do
 	esac
 done
 
-NOW=`$DATE -u +%G%m%d%k%M%S | /usr/bin/tr -d ' '`
+NOW=`$DATE -u +%G%m%d%k%M%S | /bin/sed 's/ //g'`
 if test -n "$userdir"; then
 	dir="$userdir"
 else
diff --git a/LVM2.2.02.97/scripts/fsadm.sh.old b/LVM2.2.02.97/scripts/fsadm.sh
index 4624a1c..f2b8252 100755
--- LVM2.2.02.97/scripts/fsadm.sh.old
+++ LVM2.2.02.97/scripts/fsadm.sh
@@ -32,7 +32,7 @@
 TOOL=fsadm
 
 _SAVEPATH=$PATH
-PATH=/sbin:/usr/sbin:/bin:/usr/sbin:$PATH
+PATH=/sbin:/bin:$PATH
 
 # utilities
 TUNE_EXT=tune2fs
