--- a/Configure	2013-06-14 13:27:57.803891475 +0530
+++ b/Configure	2013-06-14 13:29:46.115892219 +0530
@@ -2757,6 +2757,14 @@ LOCKF_OWNER4
 
     if test "X$LINUX_CLIB" = "X"	# {
     then
+      # Hack for ARM
+      # gnu/stubs.h includes stubs-soft.h or stubs-hard.h based on __ARM_PCS_VFP macro.
+      $CC -dM -E - < /dev/null | grep "__ARM_PCS_VFP" > /dev/null
+      if [ $? -ne 0 ] ; then
+            DEFINE_ARM_PCS_VFP=""
+      else
+            DEFINE_ARM_PCS_VFP="-D__ARM_PCS_VFP"
+      fi
       echo -n "Testing C library type with $LINUX_CONF_CC ... "
       rm -f ${LSOF_TMPC}.*
       cat > $LSOF_TMPC.c << .LSOF_END_HERE_DOC1
@@ -2771,7 +2779,9 @@ printf("\n");
 #endif
 return(0); }
 .LSOF_END_HERE_DOC1
-      $LINUX_CONF_CC ${LSOF_TMPC}.c -I$LSOF_INCLUDE -o ${LSOF_TMPC}.x > /dev/null 2>&1
+      # Intentionally using ${BUILD_CC} for compiling.
+      LINUX_CONF_CC=${BUILD_CC}
+      $LINUX_CONF_CC ${DEFINE_ARM_PCS_VFP} ${LSOF_TMPC}.c -I$LSOF_INCLUDE -o ${LSOF_TMPC}.x > /dev/null 2>&1
       if test -x ${LSOF_TMPC}.x	# {
       then
 	LINUX_CLIB=`${LSOF_TMPC}.x`
