Author:  Teddy Hogeborn <teddy@fukt.bsnet.se>
============================================================

--- netstat.c.~1~	2010-06-09 08:37:03.000000000 +0200
+++ netstat.c	2010-06-09 08:37:24.000000000 +0200
@@ -1761,12 +1761,14 @@
                  parsesnmp6(flag_raw, flag_tcp, flag_udp);
 #else
                  printf("Address type not supported for stats\n");
+                 exit(1);
 #endif
-            }
-            else
+            } else {
                  printf("Address type not supported for stats\n");
+		 exit(1);
+	    }
         }
-        exit(1);
+        exit(0);
     }
     
     if (flag_rou) {
