--- git-orig/smart/backends/rpm/base.py	2016-07-30 17:06:41.000000000 -0700
+++ git/smart/backends/rpm/base.py	2016-07-30 22:00:05.496083505 -0700
@@ -416,11 +416,21 @@
     if arch not in _sm:
         score = rpm.archscore(arch)
         _sm[arch] = score
-    return _sm.get(arch, 0)
+        for multi in ["lib64","lib32"]: 
+            if arch.startswith(multi):
+               return 2 
+        for multi in ["nativesdk"]: 
+            if arch.endswith(multi):
+               return 0 
+        return 1
 
 # TODO: Embed color into nameprovides and obsoletes relations.
 _COLORMAP = {"noarch": 0, "x86_64": 2, "ppc64": 2, "s390x": 2, "sparc64": 2}
 def getArchColor(arch, _cm=_COLORMAP):
+    for multi in ["lib64","lib32"]: 
+        if arch.startswith(multi):
+           return 1 
+    return 2
     return _cm.get(arch, 1)
 
 
