--- git/smart/backends/rpm/base.py.orig	2016-07-27 10:16:54.000000000 -0700
+++ git/smart/backends/rpm/base.py	2016-07-27 10:19:32.889550929 -0700
@@ -416,11 +416,21 @@
     if arch not in _sm:
         score = rpm.archscore(arch)
         _sm[arch] = score
-    return _sm.get(arch, 0)
+        for multi in ["lib64","lib32"]: 
+            if arch.startswith(multi):
+               return 2
+        if arch.endswith("nativesdk") and ((os.uname()[4] == "i686" and arch.startswith("x86_64")) or (os.uname()[4] == "x86_64" and arch.startswith("i686"))):
+               return 0
+
+        return 1
 
 # TODO: Embed color into nameprovides and obsoletes relations.
 _COLORMAP = {"noarch": 0, "x86_64": 2, "ppc64": 2, "s390x": 2, "sparc64": 2}
 def getArchColor(arch, _cm=_COLORMAP):
+    for multi in ["lib64","lib32"]: 
+        if arch.startswith(multi):
+           return 1
+    return 1
     return _cm.get(arch, 1)
 
 
