diff -Naurp gmp-6.0.0_org/gmp-h.in gmp-6.0.0/gmp-h.in
--- gmp-6.0.0_org/gmp-h.in	2014-03-25 07:37:55.000000000 -0700
+++ gmp-6.0.0/gmp-h.in	2016-09-06 21:18:43.462094570 -0700
@@ -40,7 +40,11 @@ see https://www.gnu.org/licenses/.  */
 #if ! defined (__GMP_WITHIN_CONFIGURE)
 #define __GMP_HAVE_HOST_CPU_FAMILY_power   @HAVE_HOST_CPU_FAMILY_power@
 #define __GMP_HAVE_HOST_CPU_FAMILY_powerpc @HAVE_HOST_CPU_FAMILY_powerpc@
-#define GMP_LIMB_BITS                      @GMP_LIMB_BITS@
+#if defined(__LP64__)
+#define GMP_LIMB_BITS                      64
+#else
+#define GMP_LIMB_BITS                      32
+#endif
 #define GMP_NAIL_BITS                      @GMP_NAIL_BITS@
 #endif
 #define GMP_NUMB_BITS     (GMP_LIMB_BITS - GMP_NAIL_BITS)
@@ -2288,8 +2292,8 @@ enum
 };
 
 /* Define CC and CFLAGS which were used to build this version of GMP */
-#define __GMP_CC "@CC@"
-#define __GMP_CFLAGS "@CFLAGS@"
+#define __GMP_CC "gcc"
+#define __GMP_CFLAGS "-O2 -pipe -g"
 
 /* Major version number is the value of __GNU_MP__ too, above and in mp.h. */
 #define __GNU_MP_VERSION            6
