--- git_org/instrumentation/events/lttng-module/sched.h	2017-05-23 20:24:33.791535459 -0700
+++ git/instrumentation/events/lttng-module/sched.h	2017-05-23 20:29:45.091919389 -0700
@@ -547,15 +547,15 @@ LTTNG_TRACEPOINT_EVENT(sched_stat_runtim
  */
 LTTNG_TRACEPOINT_EVENT(sched_pi_setprio,
 
-	TP_PROTO(struct task_struct *tsk, int newprio),
+	TP_PROTO(struct task_struct *tsk, struct task_struct *pi_task),
 
-	TP_ARGS(tsk, newprio),
+	TP_ARGS(tsk, pi_task),
 
 	TP_FIELDS(
 		ctf_array_text(char, comm, tsk->comm, TASK_COMM_LEN)
 		ctf_integer(pid_t, tid, tsk->pid)
 		ctf_integer(int, oldprio, tsk->prio - MAX_RT_PRIO)
-		ctf_integer(int, newprio, newprio - MAX_RT_PRIO)
+		ctf_integer(int, (pi_task ? pi_task->prio : tsk->prio), ((pi_task ? pi_task->prio : tsk->prio) - MAX_RT_PRIO))
 	)
 )
 #endif
