diff -Naurp git_org/Makefile git/Makefile
--- git_org/Makefile	2019-04-04 19:36:10.989772082 -0700
+++ git/Makefile	2019-04-04 19:36:38.324565192 -0700
@@ -542,8 +542,9 @@ u-boot.cfg spl/u-boot.cfg tpl/u-boot.cfg
 # gimmick, wrong config.mk would be included leading nasty warnings/errors.
 ifneq ($(wildcard $(KCONFIG_CONFIG)),)
 ifneq ($(wildcard include/config/auto.conf),)
-autoconf_is_old := $(shell find . -path ./$(KCONFIG_CONFIG) -newer \
-						include/config/auto.conf)
+autoconf_is_old := $(shell \
+		if [ "$(KCONFIG_CONFIG)" -nt "include/config/auto.conf" ] ; then \
+		echo "$(KCONFIG_CONFIG)"; fi)
 ifeq ($(autoconf_is_old),)
 include config.mk
 include arch/$(ARCH)/Makefile
