diff -Naurp parted-3.2_org/libparted/labels/dos.c parted-3.2/libparted/labels/dos.c
--- parted-3.2_org/libparted/labels/dos.c	2015-01-25 22:48:23.252411929 +0530
+++ parted-3.2/libparted/labels/dos.c	2015-01-25 23:00:09.883418092 +0530
@@ -161,6 +161,7 @@ typedef struct {
 	int		palo;
 	int		prep;
 	int		diag;
+	int             swap;
 	int		irst;
 	int		esp;
 	OrigState*	orig;			/* used for CHS stuff */
@@ -954,6 +955,7 @@ raw_part_parse (const PedDisk* disk, con
 	dos_data->raid = raw_part->type == PARTITION_LINUX_RAID;
 	dos_data->lvm = raw_part->type == PARTITION_LINUX_LVM_OLD
 			|| raw_part->type == PARTITION_LINUX_LVM;
+	dos_data->swap = raw_part->type == PARTITION_LINUX_SWAP;
 	dos_data->lba = raw_part_is_lba (raw_part);
 	dos_data->palo = raw_part->type == PARTITION_PALO;
 	dos_data->prep = raw_part->type == PARTITION_PREP;
@@ -1348,6 +1350,7 @@ msdos_partition_new (const PedDisk* disk
 		dos_data->diag = 0;
 		dos_data->raid = 0;
 		dos_data->lvm = 0;
+		dos_data->swap = 0;
 		dos_data->lba = 0;
 		dos_data->palo = 0;
 		dos_data->prep = 0;
@@ -1385,6 +1388,7 @@ msdos_partition_duplicate (const PedPart
 	new_dos_data->hidden = old_dos_data->hidden;
 	new_dos_data->raid = old_dos_data->raid;
 	new_dos_data->lvm = old_dos_data->lvm;
+	new_dos_data->swap = old_dos_data->swap;
 	new_dos_data->lba = old_dos_data->lba;
 	new_dos_data->palo = old_dos_data->palo;
 	new_dos_data->prep = old_dos_data->prep;
@@ -1436,6 +1440,7 @@ msdos_partition_set_system (PedPartition
 		dos_data->diag = 0;
 		dos_data->raid = 0;
 		dos_data->lvm = 0;
+		dos_data->swap = 0;
 		dos_data->palo = 0;
 		dos_data->prep = 0;
 		dos_data->irst = 0;
@@ -1472,6 +1477,10 @@ msdos_partition_set_system (PedPartition
 		dos_data->system = PARTITION_PREP;
 		return 1;
 	}
+	if (dos_data->swap) {
+		dos_data->system = PARTITION_LINUX_SWAP;
+		return 1;
+	}
 	if (dos_data->irst) {
 		dos_data->system = PARTITION_IRST;
 		return 1;
@@ -1519,6 +1528,7 @@ clear_flags (DosPartitionData *dos_data)
   dos_data->irst = 0;
   dos_data->esp = 0;
   dos_data->raid = 0;
+  dos_data->swap = 0;
 }
 
 static int
@@ -1580,6 +1590,12 @@ msdos_partition_set_flag (PedPartition*
 		dos_data->lvm = state;
 		return ped_partition_set_system (part, part->fs_type);
 
+	case PED_PARTITION_SWAP:
+		if (state)
+			clear_flags (dos_data);
+		dos_data->swap = state;
+		return ped_partition_set_system (part, part->fs_type);
+
 	case PED_PARTITION_LBA:
 		dos_data->lba = state;
 		return ped_partition_set_system (part, part->fs_type);
@@ -1641,6 +1657,9 @@ msdos_partition_get_flag (const PedParti
 	case PED_PARTITION_LVM:
 		return dos_data->lvm;
 
+	case PED_PARTITION_SWAP:
+		return dos_data->swap;
+
 	case PED_PARTITION_LBA:
 		return dos_data->lba;
 
@@ -1681,6 +1700,7 @@ msdos_partition_is_flag_available (const
 	case PED_PARTITION_IRST:
 	case PED_PARTITION_ESP:
 	case PED_PARTITION_DIAG:
+	case PED_PARTITION_SWAP:
 		return 1;
 
 	default:
