From 8a61d25e5240538d3e1609158232d9203c37845e Mon Sep 17 00:00:00 2001
From: Jeremy Puhlman <jpuhlman@mvista.com>
Date: Tue, 4 Dec 2018 03:02:49 +0000
Subject: [PATCH] use ls instead of run-parts

---
 client/scripts/linux | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/client/scripts/linux b/client/scripts/linux
index 1712d7d..5413ecf 100755
--- a/client/scripts/linux
+++ b/client/scripts/linux
@@ -151,7 +151,7 @@ run_hookdir() {
     dir="$1"
 
     if [ -d "$dir" ]; then
-        for script in $(run-parts --list $dir); do
+        for script in $(ls -d $dir/* 2>/dev/null); do
             run_hook $script || true
             exit_status=$?
         done
-- 
2.13.3

