--- a/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h	2013-07-23 12:00:18.845910010 +0530
+++ b/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h	2013-07-23 12:01:07.853911918 +0530
@@ -380,6 +380,10 @@
 /* CPU(AARCH64) - Aarch64 */
 #if defined(__aarch64__)
 #define WTF_CPU_AARCH64 1
+/* 32-bit mode on AARCH64 */
+#if !defined(__LP64__)
+#define WTF_CPU_AARCH64_32 1
+#endif
 #endif
 
 /* ==== OS() - underlying operating system; only to be used for mandated low-level services like 
@@ -1013,7 +1017,7 @@
     || CPU(S390X) \
     || CPU(PPC64) \
     || CPU(MIPS64) \
-    || CPU(AARCH64)
+    || CPU(AARCH64) && !CPU(AARCH64_32)
 #define WTF_USE_JSVALUE64 1
 #else
 #define WTF_USE_JSVALUE32_64 1
