diff -Naurp qt-everywhere-opensource-src-4.8.6_org/config.tests/mac/defaultarch.test qt-everywhere-opensource-src-4.8.6/config.tests/mac/defaultarch.test
--- qt-everywhere-opensource-src-4.8.6_org/config.tests/mac/defaultarch.test	2015-01-27 19:37:27.884001317 +0530
+++ qt-everywhere-opensource-src-4.8.6/config.tests/mac/defaultarch.test	2015-01-27 19:38:12.604056011 +0530
@@ -8,7 +8,7 @@ QT_MAC_DEFAULT_ARCH=
 touch defaultarch.c
 
 # compile something and run 'file' on it.
-if "$COMPILER" -c defaultarch.c 2>/dev/null 1>&2; then
+if $COMPILER -c defaultarch.c 2>/dev/null 1>&2; then
     FIlE_OUTPUT=`file defaultarch.o`
     [ "$VERBOSE" = "yes" ] && echo "'file' reports compiler ($COMPILER) default architechture as: $FIlE_OUTPUT"
 
diff -Naurp qt-everywhere-opensource-src-4.8.6_org/config.tests/mac/dwarf2.test qt-everywhere-opensource-src-4.8.6/config.tests/mac/dwarf2.test
--- qt-everywhere-opensource-src-4.8.6_org/config.tests/mac/dwarf2.test	2015-01-27 19:37:27.884001317 +0530
+++ qt-everywhere-opensource-src-4.8.6/config.tests/mac/dwarf2.test	2015-01-27 19:38:55.040063460 +0530
@@ -28,8 +28,8 @@ done
 
 touch dwarf2.c
 
-if "$COMPILER" -c dwarf2.c $CXXFLAGS -Werror -gdwarf-2 2>/dev/null 1>&2; then
-    if "$COMPILER" -c dwarf2.c $CXXFLAGS -Werror -gdwarf-2  2>&1 | grep "unsupported" >/dev/null ; then
+if $COMPILER -c dwarf2.c $CXXFLAGS -Werror -gdwarf-2 2>/dev/null 1>&2; then
+    if $COMPILER -c dwarf2.c $CXXFLAGS -Werror -gdwarf-2  2>&1 | grep "unsupported" >/dev/null ; then
         true
     else
         DWARF2_SUPPORT=yes
@@ -38,7 +38,7 @@ fi
 rm -f dwarf2.c dwarf2.o
 
 # Test for xcode 2.4.0, which has a broken implementation of DWARF
-"$COMPILER" $WORKDIR/xcodeversion.cpp $CXXFLAGS $LFLAGS -o xcodeversion -framework Carbon;
+$COMPILER $WORKDIR/xcodeversion.cpp $CXXFLAGS $LFLAGS -o xcodeversion -framework Carbon;
 ./xcodeversion
 
 if [ "$?" == "1" ]; then
diff -Naurp qt-everywhere-opensource-src-4.8.6_org/config.tests/mac/xarch.test qt-everywhere-opensource-src-4.8.6/config.tests/mac/xarch.test
--- qt-everywhere-opensource-src-4.8.6_org/config.tests/mac/xarch.test	2015-01-27 19:37:27.884001317 +0530
+++ qt-everywhere-opensource-src-4.8.6/config.tests/mac/xarch.test	2015-01-27 19:39:12.252044565 +0530
@@ -7,8 +7,8 @@ WORKDIR=$3
 
 touch xarch.c
 
-if "$COMPILER" -c xarch.c -Xarch_i386 -mmmx 2>/dev/null 1>&2; then
-    if "$COMPILER" -c xarch.c -Xarch_i386 -mmmx  2>&1 | grep "unrecognized" >/dev/null ; then
+if $COMPILER -c xarch.c -Xarch_i386 -mmmx 2>/dev/null 1>&2; then
+    if $COMPILER -c xarch.c -Xarch_i386 -mmmx  2>&1 | grep "unrecognized" >/dev/null ; then
         true
     else
         XARCH_SUPPORT=yes
