From 7ed02c41e67f03b5aa254329273f85317be2e8d3 Mon Sep 17 00:00:00 2001
From: Jeremy Puhlman <jpuhlman@mvista.com>
Date: Wed, 17 Oct 2018 21:10:29 +0000
Subject: [PATCH] No need for workaround for host cc

---
 src/Makefile.housekeeping | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.housekeeping b/src/Makefile.housekeeping
index f8334921..150ad528 100644
--- a/src/Makefile.housekeeping
+++ b/src/Makefile.housekeeping
@@ -445,7 +445,7 @@ endif
 CFLAGS		+= $(WORKAROUND_CFLAGS) $(EXTRA_CFLAGS)
 ASFLAGS		+= $(WORKAROUND_ASFLAGS) $(EXTRA_ASFLAGS)
 LDFLAGS		+= $(WORKAROUND_LDFLAGS) $(EXTRA_LDFLAGS)
-HOST_CFLAGS	+= $(WORKAROUND_CFLAGS) -O2 -g
+HOST_CFLAGS	+=  -O2 -g
 
 # Inhibit -Werror if NO_WERROR is specified on make command line
 #
-- 
2.11.1

