Index: libvirt-1.0.6/src/nodeinfo.c
===================================================================
--- libvirt-1.0.6.orig/src/nodeinfo.c
+++ libvirt-1.0.6/src/nodeinfo.c
@@ -208,9 +208,11 @@ virNodeParseSocket(const char *dir, unsi
                                  0);
 # if defined(__powerpc__) || \
     defined(__powerpc64__) || \
+    defined(__mips__) || \
+    defined(__mips64__) || \
     defined(__s390__) || \
     defined(__s390x__)
-    /* ppc and s390(x) has -1 */
+    /* ppc, mips and s390(x) has -1 */
     if (ret < 0)
         ret = 0;
 # endif
@@ -447,7 +449,9 @@ int linuxNodeInfoCPUPopulate(FILE *cpuin
              * and parsed in next iteration, because it is not in expected
              * format and thus lead to error. */
         }
-# elif defined(__arm__)
+# elif defined(__arm__) || \
+       defined(__mips__) || \
+       defined(__mips64__)
         char *buf = line;
         if (STRPREFIX(buf, "BogoMIPS")) {
             char *p;
