diff -Naurp qt-everywhere-opensource-src-4.8.6_org/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h qt-everywhere-opensource-src-4.8.6/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
--- qt-everywhere-opensource-src-4.8.6_org/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h	2015-01-27 19:45:46.143646757 +0530
+++ qt-everywhere-opensource-src-4.8.6/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h	2015-01-27 19:47:22.071559515 +0530
@@ -373,6 +373,11 @@
 #define WTF_CPU_NEEDS_ALIGNED_ACCESS 1
 #endif
 
+/* CPU(AARCH64) - Aarch64 */
+#if defined(__aarch64__)
+#define WTF_CPU_AARCH64 1
+#endif
+
 /* ==== OS() - underlying operating system; only to be used for mandated low-level services like 
    virtual memory, not to choose a GUI toolkit ==== */
 
@@ -1003,7 +1008,8 @@
     || CPU(SPARC64) \
     || CPU(S390X) \
     || CPU(PPC64) \
-    || CPU(MIPS64)
+    || CPU(MIPS64) \
+    || CPU(AARCH64)
 #define WTF_USE_JSVALUE64 1
 #else
 #define WTF_USE_JSVALUE32_64 1
