diff -ru libatomic_ops-7.4.4-orig/src/atomic_ops/sysdeps/standard_ao_double_t.h libatomic_ops-7.4.4/src/atomic_ops/sysdeps/standard_ao_double_t.h
--- libatomic_ops-7.4.4-orig/src/atomic_ops/sysdeps/standard_ao_double_t.h	2016-05-24 13:01:55.000000000 -0700
+++ libatomic_ops-7.4.4/src/atomic_ops/sysdeps/standard_ao_double_t.h	2017-03-24 10:58:37.753856212 -0700
@@ -32,7 +32,7 @@
   typedef __m128 double_ptr_storage;
 #elif defined(_WIN32) && !defined(__GNUC__)
   typedef unsigned __int64 double_ptr_storage;
-#elif defined(__aarch64__)
+#elif defined(__aarch64__) && !defined(__ILP32__)
   typedef unsigned __int128 double_ptr_storage;
 #else
   typedef unsigned long long double_ptr_storage;
Only in libatomic_ops-7.4.4/src/atomic_ops/sysdeps/: .standard_ao_double_t.h.swp
