From 5f3f558e79ec0c53bbb802b3296f3cb9f0f21705 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nyushchenko@dev.rtsoft.ru>
Date: Fri, 30 May 2014 09:22:55 +0400
Subject: [PATCH 3/3] Unconditionally flush non-flushed page data at end of
 page walk

Signed-off-by: Nikita Yushchenko <nyushchenko@dev.rtsoft.ru>
---
 kdump-tool.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/kdump-tool.c b/kdump-tool.c
index 35175c7..324c8af 100644
--- a/kdump-tool.c
+++ b/kdump-tool.c
@@ -1491,14 +1491,11 @@ process_page(struct velf_data *dpage,
 }
 
 static int
-velf_cleanup(struct elfc *pelf, struct velf_data *dpage, bool use_vaddr)
+flush_dpage(struct elfc *pelf, struct velf_data *dpage)
 {
 	int rv;
 
-	if ((use_vaddr &&
-	     ((dpage->next_vaddr - dpage->last_pgsize) != dpage->start_vaddr))
-	    || ((dpage->next_paddr - dpage->last_pgsize) != dpage->start_paddr))
-	{
+	if (dpage->prev_present) {
 		rv = gen_new_phdr(pelf, dpage);
 		if (rv)
 			return -1;
@@ -1712,7 +1709,7 @@ topelf(int argc, char *argv[])
 		}
 	}
 
-	rv = velf_cleanup(d->elf, &dpage, false);
+	rv = flush_dpage(d->elf, &dpage);
 	if (rv == -1)
 		goto out_err;
 
@@ -1979,7 +1976,7 @@ nopgd:
 	if (rv == -1)
 		goto out_err;
 
-	rv = velf_cleanup(d->elf, &dpage, true);
+	rv = flush_dpage(d->elf, &dpage);
 	if (rv == -1)
 		goto out_err;
 
-- 
1.8.3.1

