diff -uNr -x CVS /PATCH_PROCESS/packages/priors/bld-305/SOURCES/bootcycle-1.0/COPYING /PATCH_PROCESS/packages/current/bld-347/SOURCES/bootcycle-1.0/COPYING
--- bootcycle-1.0/COPYING	Wed Dec 31 16:00:00 1969
+++ bootcycle-1.0/COPYING	Thu Aug 22 13:33:13 2002
@@ -0,0 +1,2 @@
+"This software program is licensed subject to the GNU General Public License (GPL)
+Version 2, June 1991, available at <http://www.fsf.org/copyleft/gpl.html>"
diff -uNr -x CVS /PATCH_PROCESS/packages/priors/bld-305/SOURCES/bootcycle-1.0/bootcycle /PATCH_PROCESS/packages/current/bld-347/SOURCES/bootcycle-1.0/bootcycle
--- bootcycle-1.0/bootcycle	Tue Aug 13 13:23:37 2002
+++ bootcycle-1.0/bootcycle	Wed Aug 21 09:49:18 2002
@@ -91,4 +91,4 @@
 echo "[`date "+%D %T"`] System booted; auto-shutdown will occur after $REBOOTS_LEFT $REBOOT_TEXT unless system stays up for at least $BCD_MIN_UPTIME $MINUTE_TEXT" >> $LOG_FILE
 
 # execute timer script
-$WAIT_CMD $((BCD_MIN_UPTIME*60)) &
+setsid $WAIT_CMD $((BCD_MIN_UPTIME*60)) &
diff -uNr -x CVS /PATCH_PROCESS/packages/priors/bld-305/SOURCES/bootcycle-1.0/bootcycle.spec /PATCH_PROCESS/packages/current/bld-347/SOURCES/bootcycle-1.0/bootcycle.spec
--- bootcycle-1.0/bootcycle.spec	Mon Aug 19 01:51:43 2002
+++ bootcycle-1.0/bootcycle.spec	Thu Aug 29 14:36:46 2002
@@ -65,6 +65,6 @@
 # Execute the bootcycle program to detect frequent reboots
 if [ -x $BOOTCYCLE_FILE ];
 then
-    $BOOTCYCLE_FILE
+    setsid $BOOTCYCLE_FILE
 fi
 EOF
