This patch reset bootcycle detection flag in config file.
Index: bootcycle-1.0/bootcycle.conf
===================================================================
--- bootcycle-1.0.orig/bootcycle.conf
+++ bootcycle-1.0/bootcycle.conf
@@ -1,7 +1,7 @@
 # Boot Cycle Detection configuration file
 
 # yes or no, lowercase
-BCD_ENABLED=yes
+BCD_ENABLED=no
 
 # maximum number of consecutive reboots
 BCD_MAX_REBOOTS=5
