Use strip properly.

diff -r 3e2fb573db33 mvpartition/Partitioner.py
--- a/mvpartition/Partitioner.py	Fri Oct 29 17:08:51 2010 -0500
+++ b/mvpartition/Partitioner.py	Tue Mar 08 13:01:55 2011 -0600
@@ -2405,7 +2405,7 @@
         try:
             numsects = p.units.convFromStr(p, self, vals[1])
         except Exception, e:
-            p.popupWin("Size '" + strip(vals[1])
+            p.popupWin("Size '" + vals[1].strip()
                        + "' was not a valid number, please"
                        + " try again",
                        self.continueEdit, o)
