valgrind: Mask CPUID support in HWCAP on aarch64

Author: Mark Wielaard (address@hidden)

Upstream-Status: Pending

Signed-off-by: Jagadeesh Krishnanjanappa <jkrishnanjanappa@mvista.com>
diff -Naurp valgrind-3.14.0_org/coregrind/m_initimg/initimg-linux.c valgrind-3.14.0/coregrind/m_initimg/initimg-linux.c
--- valgrind-3.14.0_org/coregrind/m_initimg/initimg-linux.c	2018-07-13 14:22:05.000000000 +0530
+++ valgrind-3.14.0/coregrind/m_initimg/initimg-linux.c	2019-01-04 17:42:19.145004978 +0530
@@ -716,6 +716,12 @@ Addr setup_client_stack( void*  init_sp,
                                | VKI_HWCAP_ASIMD)
                auxv->u.a_val &= ARM64_SUPPORTED_HWCAP;
             }
+#           elif defined(VGP_arm64_linux)
+            {
+               /* Linux 4.11 started pupulating this for arm64, but we
+                  currently don't support any. */
+               auxv->u.a_val = 0;
+            }
 #           endif
             break;
 #        if defined(VGP_ppc64be_linux) || defined(VGP_ppc64le_linux)
