--- libvirt-1.0.0_org/configure.ac	2013-06-01 19:11:00.724656674 +0530
+++ libvirt-1.0.0/configure.ac	2013-06-01 19:15:29.140665794 +0530
@@ -958,6 +958,46 @@ dnl
 dnl check for kernel headers required by src/bridge.c
 dnl
 if test "$with_qemu" = "yes" || test "$with_lxc" = "yes" ; then
+  # We need linux/in6.h header file included before linux/if_bridge.h.
+  AC_CHECK_HEADERS([linux/in6.h])
+  AC_CHECK_HEADERS([linux/if_bridge.h], [], [], 
+  [#include <stdio.h>
+  #ifdef HAVE_SYS_TYPES_H
+  # include <sys/types.h>
+  #endif
+  #ifdef HAVE_SYS_STAT_H
+  # include <sys/stat.h>
+  #endif
+  #ifdef STDC_HEADERS
+  # include <stdlib.h>
+  # include <stddef.h>
+  #else
+  # ifdef HAVE_STDLIB_H
+  #  include <stdlib.h>
+  # endif
+  #endif
+  #ifdef HAVE_STRING_H
+  # if !defined STDC_HEADERS && defined HAVE_MEMORY_H
+  #  include <memory.h>
+  # endif
+  # include <string.h>
+  #endif
+  #ifdef HAVE_STRINGS_H
+  # include <strings.h>
+  #endif
+  #ifdef HAVE_INTTYPES_H
+  # include <inttypes.h>
+  #endif
+  #ifdef HAVE_STDINT_H
+  # include <stdint.h>
+  #endif
+  #ifdef HAVE_UNISTD_H
+  # include <unistd.h>
+  #endif
+  #ifdef HAVE_LINUX_IN6_H
+  # include <linux/in6.h>
+  #endif
+  ])
   AC_CHECK_HEADERS([linux/param.h linux/sockios.h linux/if_bridge.h linux/if_tun.h],,
                    AC_MSG_ERROR([You must install kernel-headers in order to compile libvirt with QEMU or LXC support]))
 fi
--- libvirt-1.0.0_org/src/util/virnetdevbridge.c	2012-09-21 07:32:13.000000000 +0530
+++ libvirt-1.0.0/src/util/virnetdevbridge.c	2013-06-01 19:09:53.900654070 +0530
@@ -36,6 +36,7 @@
 #ifdef __linux__
 # include <linux/sockios.h>
 # include <linux/param.h>     /* HZ                 */
+# include <linux/in6.h>       /* Defines struct in6_addr */
 # include <linux/if_bridge.h> /* SYSFS_BRIDGE_ATTR  */
 
 # define JIFFIES_TO_MS(j) (((j)*1000)/HZ)
