Index: lxc-0.7.5/src/lxc/conf.c
===================================================================
--- lxc-0.7.5.orig/src/lxc/conf.c
+++ lxc-0.7.5/src/lxc/conf.c
@@ -1602,10 +1602,6 @@ void lxc_delete_network(struct lxc_list
 		if (netdev->ifindex == 0)
 			continue;
 
-		/* Recent kernels already delete the virtual devices */
-		if (netdev->type != LXC_NET_PHYS)
-			continue;
-
 		if (lxc_netdev_rename_by_index(netdev->ifindex, netdev->link))
 			WARN("failed to rename to the initial name the netdev '%s'",
 			     netdev->link);
