Index: lxc-0.7.4.1/doc/lxc.conf.sgml.in
===================================================================
--- lxc-0.7.4.1.orig/doc/lxc.conf.sgml.in	2011-04-27 10:43:36.000000000 -0700
+++ lxc-0.7.4.1/doc/lxc.conf.sgml.in	2011-04-27 10:44:32.000000000 -0700
@@ -100,6 +100,10 @@
 	      <option>i686</option>,
 	      <option>x86_64</option>,
 	      <option>amd64</option>
+	      <option>ppc</option>
+	      <option>ppc64</option>
+	      <option>mips</option>
+	      <option>mips64</option>
 	    </para>
 	  </listitem>
 	</varlistentry>
Index: lxc-0.7.4.1/src/lxc/confile.c
===================================================================
--- lxc-0.7.4.1.orig/src/lxc/confile.c	2011-04-27 10:45:43.000000000 -0700
+++ lxc-0.7.4.1/src/lxc/confile.c	2011-04-27 10:46:18.000000000 -0700
@@ -509,11 +509,15 @@
 	struct per_name {
 		char *name;
 		int per;
-	} pername[4] = {
+	} pername[8] = {
 		{ "x86", PER_LINUX32 },
 		{ "i686", PER_LINUX32 },
 		{ "x86_64", PER_LINUX },
 		{ "amd64", PER_LINUX },
+		{ "ppc", PER_LINUX32 },
+		{ "ppc64", PER_LINUX },
+		{ "mips", PER_LINUX32 },
+		{ "mips64", PER_LINUX },
 	};
 
 	int i;
